shared/src/types/ethereum_events.rs

```diff
@@ -55,6 +59,27 @@ impl From<u64> for Uint {
 )]
 pub struct EthAddress(pub [u8; 20]);
 
+impl EthAddress {
+    /// The canonical way we represent an [`EthAddress`] in storage keys. A
+    /// 40-character lower case hexadecimal address prefixed by '0x'.
+    /// e.g. "0x6b175474e89094c44da98b954eedeac495271d0f"
+    pub fn to_canonical(&self) -> String {
+        format!("{:?}", ethabi::ethereum_types::Address::from(&self.0))
+    }
+}
```

Comment:
We could alternatively `impl Display` here though I thought this would be more explicit